src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/request/RaptorRoutingRequestTransitDataCreator.java

```diff
@@ -85,9 +85,10 @@
   ) {
 
     // Group TripPatternForDate objects by TripPattern
-    Map<TripPatternWithRaptorStopIndexes, List<TripPatternForDate>> patternForDateByPattern = patternForDateList
-        .stream()
-        .collect(groupingBy(TripPatternForDate::getTripPattern));
+    Map<TripPatternWithRaptorStopIndexes, List<TripPatternForDate>> patternForDateByPattern = new HashMap<>();
```

Comment:
This need a bit of doc on why we are using a for loop and not a stream so we avoid that someone changes it back to a stream later. But, the findings that stream should be avoided in high performance use case is very intresting.

Reply:
Added